### DISCUSSION TITLE
Custom server endpoint configuration

### DISCUSSION THREAD

**Discussion by JustRoxy** [ppy/osu#18998] [Ideas] [answered]
Add feature to override production API endpoints.

Right now there's no possibility to change endpoints to custom osu! server without rebuilding the whole game.
It would be very convenient to change them in config file/settings without shipping of custom client with overwritten API routes.

**Comment by bdach** [comment]
Why would this be needed?

- If you want to test osu! full-stack, you can already, by changing source. [There is even a how to guide about this.](https://github.com/ppy/osu/wiki/Testing-web---server-full-stack-with-osu!)
- If this is to facilitate "custom servers", then this isn't and won't be a supported use case. See: https://github.com/ppy/osu/discussions/13822 (and I could have sworn this was brought up and rejected on several other occasions)

**Comment by peppy** [comment]
As per discord conversation, please state *why* this is required. Why do you need a custom server or custom client in the first place? If you have a reason, there is a high chance that can be resolved without doing what you are trying to do.

**Reply by JustRoxy** [reply]
Custom servers will always be created as history of stable tells. 
Addition of the feature will remove harm of downloading and running third-party programs or clients that will mock/strip osu! certificate with redirection to custom endpoints.

**Reply by peppy** [reply, marked as answer]
an eventual plan would be to better support such cases, but it's *very* low priority. understand that we are working on this game to support the full ecosystem. for now, the advice i gave in the other thread remains: consider making your own game. lazer is in active development and creating custom server is counterproductive in every way.

**Reply by GeeTwentyFive** [reply]
> As per discord conversation, please state _why_ this is required. Why do you need a custom server or custom client in the first place? If you have a reason, there is a high chance that can be resolved without doing what you are trying to do.

Akatsuki; Relax is unranked in osu!'s official server, but ranked in Akatsuki. There is a solution for osu!stable, but stable has notelock. With osu!Lazer support, one could enjoy ranked Relax without notelock.
